docs/faq.rst

```diff
@@ -493,6 +493,37 @@ maybe open an issue in their issue tracker. Do not file an issue in the
 If you can reproduce the issue with the proven filesystem, please file an
 issue in the |project_name| issue tracker about that.
 
+Why does running 'borgbackup check --repair' warn about data loss?
+------------------------------------------------------------------
+
+Due to the nature of backup, and especially deduplicated backups as used
+by |project_name|, some repair operations may require a roll-back to an
+earlier version of the repository. Some errors, especially where data
+corruption is involved, may require purging corrupted data to bring the
+repository back to a known good state.
+
+Also, we can't predict every possible corruption vector for a repository.
+Some forms of corruption may be undetected, and some attempts to repair
+corruption could break other parts of the repository.
+
+Repair is considered a destructive operation in that it will modify the
+repository in order to automatically repair some known methods of corruption.
+
+In other words, the warning is there to emphasize that |project_name| will:
+  - Perform automated routines that will modify your backup repository
+  - May not actually fix the problem you are experiencing
+  - May further corrupt your repository
+  - May cause other problems which may not be noticable
+  - May cause contents of backed up data to become corrupted or changed
+
+Aside from this, if you find yourself in a position where you believe your
+repository is corrupt, this operation may result in recovering at least
+some of your backup data. You are **STRONGLY ENCOURAGED** to attempt the
+repair on a copy of the repo in case the repository becomes more unusable
+after the repair.
+
+If you are still unsure, seek :ref:`support`!
+
```

Comment:
I'll review all of this.  I think this FAQ got a little long and definitely comes across negatively, which wasn't my intention.  I wanted to try to make clear why the warning existed, and to explain "What's the worst that could happen if I run --repair?"  The hope being, that a reader would be more likely to take the time to read the part about repairing a repo copy. Especially the last line about seeking support if they need help.

You have put a lot of time into my pull requests - I appreciate it.  My intention was to save you the time and trouble of having to write these things. I hope I haven't wasted your time ... that was not my intention at all.

I did these tasks because it helps me dive into the code. I try not to write things which aren't true, so I try to review the code and existing documentation where possible to create informed documentation.

At any  rate, thank you for your time and attention - hopefully I can make some fixes which will resolve your concerns. :)